flink-streaming-java/src/main/java/org/apache/flink/streaming/api/graph/StreamEdge.java

```diff
@@ -60,17 +60,23 @@
 	 */
 	private StreamPartitioner<?> outputPartitioner;
 
+	/**
+	 * The unique id for differentiating edges between the same source and target.
+	 */
+	private final int edgeSubId;
+
 	public StreamEdge(StreamNode sourceVertex, StreamNode targetVertex, int typeNumber,
-			List<String> selectedNames, StreamPartitioner<?> outputPartitioner, OutputTag outputTag) {
+			List<String> selectedNames, StreamPartitioner<?> outputPartitioner, OutputTag outputTag, int edgeSubId) {
 		this.sourceVertex = sourceVertex;
 		this.targetVertex = targetVertex;
 		this.typeNumber = typeNumber;
 		this.selectedNames = selectedNames;
 		this.outputPartitioner = outputPartitioner;
 		this.outputTag = outputTag;
+		this.edgeSubId = edgeSubId;
```

Comment:
rename `edgeSubId` to `uniqueId`? Since it's not a "sub" id of the `edgeId`, but rather a component of it.